pyccel/parser/semantic.py

```diff
@@ -2131,27 +2131,32 @@ def _visit_Assign(self, expr, **settings):
             else:
 
                 # TODO check types from FunctionDef
-
                 master = macro.master
                 name = _get_name(master.name)
-
-                # all terms in lhs must be already declared and available
-                # the namespace
-                # TODO improve
+                results = []
+                args = [self._visit(i, **settings) for i in rhs.args]
+                args_names = [arg.name for arg in args if isinstance(arg, Variable)]
 
                 if not sympy_iterable(lhs):
                     lhs = [lhs]
+                results_shapes = macro.get_results_shapes(args)
+                for m_result, shape, result in zip(macro.results, results_shapes, lhs):
+                    if m_result in macro.master_arguments and not result in args_names:
+                        index = macro.master_arguments.index(m_result)
+                        d_result = self._infere_type(master.arguments[index])
+                        d_result['shape'] = shape
+                        tmp = self._assign_lhs_variable(result, d_result, None, new_expressions, False, **settings)
+                        results.append(tmp)
+                    elif result in args_names:
+                        _name = _get_name(result)
+                        tmp = self.get_variable(_name)
+                        results.append(tmp)
+                    else:
+                        # TODO: check for result in master_results
```

Comment:
Is this TODO still valid?

Reply:
I just added this TODO, I though we might want to have master_results in macros, this is suggested in another TODO

Reply:
ok :+1: